ocaml/xapi/stream_vdi.ml

```diff
@@ -100,6 +100,53 @@ let write_block ~__context filename buffer ofd len =
     then raise (Api_errors.Server_error (Api_errors.client_error, [ExnHelper.string_of_exn e]))
     else raise e
 
+(* Example = [{"length": 2097152, "flags": 0}, {"length": 1902848, "flags": 3}]*)
+type descriptor = {
+  length : int64;
+  status_flags : int32;
+}
+
+let cycle_descriptors descriptor_list offset =
+  let rec range a b =
+    if a > b then []
+    else a :: range (a + 1) b
+  in
+  (* Insert proper not_empty function here *)
+  let not_empty status_flag =
+    match (Int32.to_int status_flag) with
+      | 3 -> true
+      | _ -> false
+  in
+  let find_blocks descriptor offset increment =
+    match (not_empty descriptor.status_flags) with
+        | true -> begin
+            let start_chunk = offset / increment in
+            (* If the chunk ends at the boundary don't include the next chunk *)
+            let end_chunk = (Int64.to_int descriptor.length + offset - 1) / increment in
```

Comment:
Ah I see, so this works because chunks start at `0`

Reply:
Indentation looks too deep for the cases. It's a bit unusual to use match rather than if for boolean expressions but it results it a nice table-like layout.

Reply:
Ah looks like a have used a mix of 2 and 4 space identations. What should it be? Would you prefer I swap to if/else?

Reply:
2 spaces. We use the ocp-indent program to reindent files (but that might change other parts of the file too)